From 7241c1e2f7c175167d1046e704f06f73b093ed0b Mon Sep 17 00:00:00 2001
From: Devi Priya <quic_devipriy@quicinc.com>
Date: Fri, 13 Sep 2024 20:11:22 +0530
Subject: [PATCH] serial: qcom-geni: Update the default timeout in
 geni_serial_poll_bit

The default timeout value in qcom_geni_serial_poll_bit is not sufficient
for flushing the serial data when enabling earlycon.
Update the default timeout value calculated based on the below formula
timeout_us = ((fifo_bits * USEC_PER_SEC) / baud) + 500;
Where fifo_bits = fifo_depth * fifo_width * 32

Change-Id: Ib2a03bdc04325a9f2b99d341e28c6d582612302c
Signed-off-by: Devi Priya <quic_devipriy@quicinc.com>
---
 drivers/tty/serial/qcom_geni_serial.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/tty/serial/qcom_geni_serial.c b/drivers/tty/serial/qcom_geni_serial.c
index 41a15c500963..05ab1bffe206 100644
--- a/drivers/tty/serial/qcom_geni_serial.c
+++ b/drivers/tty/serial/qcom_geni_serial.c
@@ -277,7 +277,7 @@ static bool qcom_geni_serial_poll_bit(struct uart_port *uport,
 	struct qcom_geni_serial_port *port;
 	unsigned int baud;
 	unsigned int fifo_bits;
-	unsigned long timeout_us = 20000;
+	unsigned long timeout_us = 200000;
 	struct qcom_geni_private_data *private_data = uport->private_data;
 
 	if (private_data->drv) {
-- 
2.34.1

